### PR TITLE
Replace Sun Jersey Client with Glassfish Jersey Client

### PR DESCRIPTION
**Pull request by csvtuda**
This PR replaces the deprecated [Sun Jersey Client](https://mvnrepository.com/artifact/com.sun.jersey/jersey-client) with the [Glassfish Jersey Client](https://mvnrepository.com/artifact/org.glassfish.jersey.core/jersey-client).

### CODE REVIEW

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                            )
+                    );
+                }
+            }
+        }
+        QtafFactory.getLogger().error(String.format("[QTAF HTTP] Request %s failed.", request));
+        return null;
+    }
+
+    public static RequestBuilder buildRequest(URI uri) {
```
Comment (qytera-development) [inline qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java:51]: Please add Docstring

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        WebTarget target = CLIENT.target(uri);
+        HTTPEvents.webResourceAvailable.onNext(target);
+        return new RequestBuilder(uri, target.request());
+    }
+
+    public static Response post(RequestBuilder requestBuilder, JsonElement body) {
```
Comment (qytera-development) [inline qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java:57]: Please add Docstring

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return wrapInRetry(
+                requestBuilder.getBuilder().buildPost(Entity.entity(body, MediaType.APPLICATION_JSON_TYPE)),
+                requestBuilder.getPath()
+        );
+    }
+
+    public static Response get(RequestBuilder requestBuilder) {
```
Comment (qytera-development) [inline qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java:64]: Please add Docstring

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+/**
+ * HTTP Dao Test
+ */
+public class HTTPDaoTest {
+    /**
+     * Host we are testing against
+     */
+    final String HOST = "http://jsonplaceholder.typicode.com";
+
+    @Test
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:17]: Please add a comment to the test case

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                  "postId": 1,
+                  "id": 1,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""");
+    }
+
+    @Test
+    public void testHttpGetRequestWithResponseObject() {
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:32]: Please add a comment to the test case

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                  "postId": 1,
+                  "id": 1,
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""");
+    }
+
+    @Test
+    public void testHttpPostRequestWithStringMapping() {
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:48]: Please add a comment to the test case

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""";
+        try (Response result = httpDao.post("/comments", MediaType.APPLICATION_JSON, newCommentEntity)) {
+            Assert.assertEquals(result.getStatus(), 201);
+            String json = result.readEntity(String.class);
+            Assert.assertEquals(json, newCommentEntity);
+        }
+    }
+
+    @Test
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:80]: Please add a comment to the test case

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                  "name": "id labore ex et quam laborum",
+                  "email": "john.doe@inet.com",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""";
+        HTTPDao httpDao = new HTTPDao(HOST);
+        String result = httpDao.putAsString("/comments/1", MediaType.APPLICATION_JSON, newCommentEntity);
+        Assert.assertEquals(result, newCommentEntity);
+    }
+
+    @Test
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:95]: Please add a comment to the test case

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                }""";
+        String json;
+        try (Response result = httpDao.put("/comments/1", MediaType.APPLICATION_JSON, updatedCommentEntity)) {
+            Assert.assertEquals(result.getStatus(), 200);
+            json = result.readEntity(String.class);
+            Assert.assertEquals(json, updatedCommentEntity);
+        }
+    }
+
+    @Test
```
Comment (qytera-development) [inline qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:114]: Please add a comment to the test case

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+     *
+     * @return the base URL
+     */
+    default String getXrayURL() {
+        if (XrayConfigHelper.isXrayCloudService()) {
+            return XrayRestPaths.XRAY_CLOUD_API_V2;
+        }
+        return XrayConfigHelper.getServerUrl();
+    }
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java:36]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.Map;
+
+public interface XrayTestRepository extends XrayEndpoint {
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java:25]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+    static XrayTestRepository getInstance() {
+        return XrayConfigHelper.isXrayCloudService() ? new XrayTestRepositoryCloud() : new XrayTestRepositoryServer();
+    }
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java:29]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+import jakarta.ws.rs.core.Response;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:23]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+                            )
+                    );
+                    break;
+                }
+            }
+        }
+        // Unfortunately, Xray only returns issue IDs instead of issue keys. Therefore, we need this extra step now.
+        return convertIssueIdsToIssueKeys(stepsByIssue, testIssueKeys);
+    }
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:101]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+    private URI getURIGraphQL() throws URISyntaxException {
+        return new URI(getXrayURL() + "/graphql");
+    }
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:105]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+                    stepResponse.errorReason()
+            ));
+            return Collections.emptyMap();
+        }
+        Map<String, XrayTestStepResponseDto[]> stepsByIssue = new HashMap<>();
+        Arrays.stream(stepResponse.getData().getGetTests().getResults())
+                .forEach(r -> stepsByIssue.put(r.getIssueId(), r.getSteps()));
+        return stepsByIssue;
+    }
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:124]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+import jakarta.ws.rs.core.Response;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java:22]: Please add DocString

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+                        responseData,
+                        XrayServerTestStepResponseDto[].class
+                );
+                stepsByIssue.put(testIssueKey, steps);
+            }
+        }
+        return stepsByIssue;
+    }
+
+
```
Comment (qytera-development) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java:63]: Please add DocString

**Review by qytera-development — COMMENTED** [review]
Review added. Please add DocStrings to classes and methods.

File: qtaf-http/src/main/java/de/qytera/qtaf/http/RequestBuilder.java [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.net.URI;
+
+@Getter
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public class RequestBuilder {
+
+    private final URI path;
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/RequestBuilder.java:24]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/main/java/de/qytera/qtaf/http/RequestBuilder.java [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+    private final Invocation.Builder builder;
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/RequestBuilder.java:28]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+     * Jersey HTTP Client.
+     */
+    private static final Client CLIENT = ClientBuilder.newClient().register(JsonGsonProvider.class);
+
+    /**
+     * Maximum number of retries in case an HTTP request fails because of networking issues.
+     */
+    private static final int MAX_RETRIES = 3;
+
+    private static Response wrapInRetry(Invocation request, URI path) {
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java:43]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java (hunk 1 above)
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java:75]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java (hunk 2 above)
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java (hunk 14 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:101]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java (hunk 3 above)
Comment (csvtuda) [reply qtaf-http/src/main/java/de/qytera/qtaf/http/WebService.java]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java (hunk 4 above)
Comment (csvtuda) [reply qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:17]: I am not the author of this test case, I'm not quite sure what to properly comment on.

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java (hunk 7 above)
Comment (csvtuda) [reply qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:80]: I am not the author of this test case, I'm not quite sure what to properly comment on.

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+                  "name": "id labore ex et quam laborum",
+                  "email": "Eliseo@gardner.biz",
+                  "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\\ntempora quo necessitatibus\\ndolor quam autem quasi\\nreiciendis et nam sapiente accusantium"
+                }""";
+        HTTPDao httpDao = new HTTPDao(HOST);
+        String result = httpDao.postAsString("/comments", MediaType.APPLICATION_JSON, newCommentEntity);
+        Assert.assertEquals(result, newCommentEntity);
+    }
+
+    @Test
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:62]: I am not the author of this test case, I'm not quite sure what to properly comment on.

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java (hunk 6 above)
Comment (csvtuda) [reply qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:48]: I am not the author of this test case, I'm not quite sure what to properly comment on.

File: qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java (hunk 5 above)
Comment (csvtuda) [reply qtaf-http/src/test/java/de/qytera/qtaf/http/HTTPDaoTest.java:32]: I am not the author of this test case, I'm not quite sure what to properly comment on.

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/events/XrayEvents.java [hunk 23]
```diff
@@ -0,0 +1,41 @@
+package de.qytera.qtaf.xray.events;
+
+import de.qytera.qtaf.xray.dto.request.xray.ImportExecutionResultsRequestDto;
+import de.qytera.qtaf.xray.dto.response.xray.ImportExecutionResultsResponseDto;
+import jakarta.ws.rs.core.Response;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import rx.subjects.BehaviorSubject;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/events/XrayEvents.java:10]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java [hunk 24]
```diff
@@ -0,0 +1,51 @@
+package de.qytera.qtaf.xray.repository.jira;
+
+import de.qytera.qtaf.core.QtafFactory;
+import de.qytera.qtaf.core.config.exception.MissingConfigurationValueException;
+import de.qytera.qtaf.core.util.Base64Helper;
+import de.qytera.qtaf.xray.config.XrayConfigHelper;
+
+public interface JiraEndpoint {
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java:11]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+     * @throws MissingConfigurationValueException if no Jira username or API token have been configured
+     */
+    default String getJiraAuthorizationHeaderValue() throws MissingConfigurationValueException {
+        if (XrayConfigHelper.isXrayCloudService()) {
+            return getHeaderCloud();
+        }
+        return getHeaderServer();
+    }
+
+    private static String getHeaderCloud() throws MissingConfigurationValueException {
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java:26]: It's a private method, I think it does not require a JavaDoc string.

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+        String apiToken = XrayConfigHelper.getAuthenticationJiraAPIToken();
+        if (apiToken == null) {
+            throw new MissingConfigurationValueException(
+                    XrayConfigHelper.AUTHENTICATION_JIRA_API_TOKEN,
+                    QtafFactory.getConfiguration());
+        }
+        String encoded = Base64Helper.encode(String.format("%s:%s", username, apiToken));
+        return String.format("Basic %s", encoded);
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraEndpoint.java:43]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java [hunk 27]
```diff
@@ -0,0 +1,113 @@
...
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class JiraIssueRepository implements JiraEndpoint {
+
+    private static final JiraIssueRepository INSTANCE = new JiraIssueRepository();
+
+    public static JiraIssueRepository getInstance() {
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java:36]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java [hunk 28]
```diff
@@ -0,0 +1,113 @@
...
+                        responseData,
+                        JiraIssueSearchResponseDto.class
+                );
+                total = paginatedResult.getTotal();
+                issues.addAll(Arrays.asList(paginatedResult.getIssues()));
+            }
+        }
+        return issues;
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java:95]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+import de.qytera.qtaf.xray.events.XrayEvents;
+import de.qytera.qtaf.xray.log.XrayAuthenticationErrorLog;
+import jakarta.ws.rs.core.Response;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java:22]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java [hunk 30]
```diff
@@ -0,0 +1,113 @@
...
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/jira/JiraIssueRepository.java:22]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class XrayCloudAuthenticator {
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java:28]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java [hunk 32]
```diff
@@ -0,0 +1,72 @@
...
+    private static String JWT_TOKEN;
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java:30]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+                    ErrorLogCollection.getInstance().addErrorLog(authErrorLog);
+                } else {
+                    XrayEvents.authenticationSuccess.onNext(true);
+                    JWT_TOKEN = responseData;
+                }
+            }
+        }
+        return String.format("Bearer %s", JWT_TOKEN);
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCloudAuthenticator.java:63]: It's a private method, so I think JavaDoc is not necessary.

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCucumberRepositoryCloud.java [hunk 34]
```diff
@@ -65,48 +57,49 @@ public String getFeatureFileDefinition(String[] testIDs) {
...
+        RequestBuilder request = WebService.buildRequest(getURIExportFeatureFiles(keys));
+        request.getBuilder()
+                .header(HttpHeaders.AUTHORIZATION, getXrayAuthorizationHeaderValue())
+                .accept("application/zip");
+        try (Response response = WebService.get(request)) {
+            InputStream inputStream = response.readEntity(InputStream.class);
+            return new ZipInputStream(inputStream);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCucumberRepositoryCloud.java:75]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java (hunk 18 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java:66]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java (hunk 17 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryServer.java:22]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java (hunk 13 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java:26]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java (hunk 11 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java:29]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java [hunk 35]
```diff
@@ -0,0 +1,87 @@
...
+import de.qytera.qtaf.xray.events.XrayEvents;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.Map;
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepository.java:23]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java [hunk 36]
```diff
@@ -0,0 +1,51 @@
...
+    private static String getHeaderCloud() throws MissingConfigurationValueException, URISyntaxException {
+        return XrayCloudAuthenticator.getXrayAuthorizationHeaderValue();
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java:41]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java (hunk 10 above)
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java:37]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java [hunk 37]
```diff
@@ -0,0 +1,51 @@
+package de.qytera.qtaf.xray.repository.xray;
+
+import de.qytera.qtaf.core.QtafFactory;
+import de.qytera.qtaf.core.config.exception.MissingConfigurationValueException;
+import de.qytera.qtaf.xray.config.XrayConfigHelper;
+import de.qytera.qtaf.xray.config.XrayRestPaths;
+
+import java.net.URISyntaxException;
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java]: Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCucumberRepositoryServer.java [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+    public String getFeatureFileDefinition(String[] testIDs) throws URISyntaxException, MissingConfigurationValueException {
+        RequestBuilder request = WebService.buildRequest(getURIExportFeatureFiles(testIDs));
+        request.getBuilder()
+                .header(HttpHeaders.AUTHORIZATION, getXrayAuthorizationHeaderValue())
+                .accept(MediaType.APPLICATION_JSON_TYPE);
+        try (Response response = WebService.get(request)) {
+            return response.readEntity(String.class);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (csvtuda) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayCucumberRepositoryServer.java:29]: Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).

**Review by mzdih — APPROVED** [review]
PR nicht reviewed, nur approved

**Pull request merged**